src/components/RedeemPopup.tsx

```diff
@@ -0,0 +1,32 @@
+import { BottomPaper } from "@creativesolutionsgroup/bottom-paper"
+import { Button, Paper, Typography } from "@mui/material"
+import { PropsWithChildren, useEffect, useState } from "react"
+import { IReward } from "../models/User"
+
+interface PopupProps { reward: IReward, show?: boolean, onClose: Function };
+
+export const RedeemPopup = ({ reward, show, onClose }: PropsWithChildren<PopupProps>) => {
+  // The last Y scroll position.
+  // This helps us set the fixed position bug.
+  const [last_scroll, set_last_scroll] = useState(window.scrollY);
+
+  useEffect(() => {
+    let scroll = show ? window.scrollY : last_scroll;
+    set_last_scroll(scroll);
+    document.body.style.top = show ? `-${window.scrollY}px` : "";
+
+    document.body.style.position = show ? "fixed" : "";
+    document.body.style.overflow = show ? "hidden" : "scroll";
+    document.body.style.overscrollBehaviorY = show ? "none" : "";
+    !show && window.scrollTo(0, scroll);
```

Comment:
I am unclear on the purpose of line 21. Invert the value of show and allow the window to scroll from 0 to scroll which is equal to show, or if undefined, window.scrollY which is the default value of last_scroll's useState??